pkg/xds/generator/outbound_proxy_generator.go

```diff
@@ -27,142 +29,172 @@ func (g OutboundProxyGenerator) Generate(ctx xds_context.Context, proxy *model.P
 		return nil, nil
 	}
 	resources := &model.ResourceSet{}
-	sourceService := proxy.Dataplane.Spec.GetIdentifyingService()
-	meshName := ctx.Mesh.Resource.GetMeta().GetName()
-	ofaces, err := proxy.Dataplane.Spec.Networking.GetOutboundInterfaces()
-	if err != nil {
-		return nil, err
-	}
-	for i, outbound := range outbounds {
-		// pick a route
-		route := proxy.TrafficRoutes[outbound.Service]
-		if route == nil {
-			return nil, errors.Errorf("%s{service=%q}: has no TrafficRoute", validators.RootedAt("dataplane").Field("networking").Field("outbound").Index(i), outbound.Service)
-		}
+	allClusters := envoy_common.Clusters{}
 
-		// determine the list of destination clusters
-		clusters, err := g.determineClusters(ctx, proxy, route)
+	for _, outbound := range outbounds {
+		// Determine the list of destination clusters
+		// For one outbound listener it may contain many clusters (ex. TrafficRoute to many destinations)
+		clusters, err := g.determineClusters(proxy, outbound)
```

Comment:
`clusters` and `allClusters` are not mapped to Envoy clusters 1 to 1, maybe make sense to consider renaming to `subset` or something like that

Reply:
Ok, changed ClusterInfo to ClusterSubsets. and ClusterInfo#Name to ClusterSubset#ClusterName. I think it make more sense now.